Sources/Apodini/Request/Request.swift

```diff
@@ -6,29 +6,151 @@
 //
 import Foundation
 import NIO
-import protocol Fluent.Database
+import protocol FluentKit.Database
+import struct FluentKit.DatabaseID
+
+protocol ApodiniRequest: CustomStringConvertible, CustomDebugStringConvertible {
+    /// Returns a description of the Apodini Request.
+    /// If the `ExporterRequest` also conforms to `CustomStringConvertible`, its `description`
+    /// will be appended.
+    var description: String { get }
+    /// Returns a debug description of the Apodini Request.
+    /// If the `ExporterRequest` also conforms to `CustomDebugStringConvertible`, its `debugDescription`
+    /// will be appended.
+    var debugDescription: String { get }
+
+    var endpoint: AnyEndpoint { get }
 
-/// A type representing a server response.
-public protocol Request: CustomStringConvertible {
-    /// parameters
-    func parameter<T: Codable>(for parameter: UUID) throws -> T?
-    /// The EventLoop associated with this request
     var eventLoop: EventLoop { get }
-    /// database
-    var database: Fluent.Database? { get }
+
+    var defaultDatabase: Database? { get }
+    func database(_ id: DatabaseID?) -> Database?
+
+    func retrieveParameter<Element: Codable>(_ parameter: Parameter<Element>, of element: Element.Type) throws -> Element
```

Comment:
I do not understand the use of the argument `of element: Element.Type`.  Why is this needed?

Reply:
Oh, I See. Might be some leftover parameter from a previous state where the generic wasn't used with the Parameter.
This can indeed be removed.

Reply:
Solved via 37ebdfa